article.md

```diff
@@ -0,0 +1,322 @@
+Elements of the domain are as follows:
+* `name` - the user readable name of signing domain, i.e. the name of the DApp or the protocol.
+* `version` - the current major version of the signing domain. Signatures from different versions are not compatible.
+* `contractAddress` - the address of the contract that will verify the signature. It ensures that the signature can only be used by our contract on the given chain.
+
+`Payload` will be the main structure that we'll be constructing and signing. For the sake of simplicity, it only has one field but the EIP-712 standard offers a lot of flexibility.
+
+
+## Signing with an Ethereum wallet
+Let's use Metamask as an example. You just need to call the `signTypedData_v4` ffunction with proper parameters and it will return the signature that we need.<br/>
+```js
+const msg = {
+  domain: {
+    name: "Confirm the ownership of this StarkNet address",
+    verifyingContract: '0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF',
+    version: '1',
+  },
+  message: {
+    "starknetAddress": 12356789,
+  },
+  primaryType: 'Payload',
+  types: {
+    domain: [
+      { name: 'name', type: 'string' },
+      { name: 'verifyingContract', type: 'address' },
+      { name: 'version', type: 'string' },
+    ],
+    Payload: [
+      { name: 'starknetAddress', type: 'uint256' },
+    ],
+  },
+};
+const accounts = await window.ethereum.request({method: "eth_requestAccounts"})
+await window.ethereum.request({
+    method: "eth_signTypedData_v4",
+    from: accounts[0],
+    params: [accounts[0], JSON.stringify(msg)]
+})
+```
+After you run this this JS code in the browser's console on the MetaMask website it will trigger a signing request. An application would use a similar code to get user's accounts and signature. <br/><br/>
+<img alt="Screenshot 2022-10-12 at 10 41 54" src="./graphics/metamask.png"><br/><br/>
+As you can see everything is transparent so you don't have to sign something blindly - that's one of the pros of using EIP-712 standard.<br/><br/>
+<img alt="Screenshot 2022-10-12 at 10 41 54" src="./graphics/signature.png"><br/>
+When you sign this message you will receive the message signature that we need. After that we can send a transaction containing both the Ethereum address and the signature to StarkNet's contract.
+## Creating a connection
+Before we create a connection we need to ensure that the caller owns both the addresses. This is how we can do it:
+1. Sign the structure containing the StarkNet address with our Ethereum account
+2. Call the contract that we are creating with received signature and our Ethereum address from your StarkNet account.
+3. By calling `get_caller_address()` and using it as our Starknet address we ensure that it indeed belongs to the user and we will use it when reconstructing a signed message.
+4. Recreate the message structure and hash it with keccak256.
```

Comment:
I think point 3 is an implementation detail and can be omitted, but we should mention in next point that the contract handles that.